source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -111,12 +111,26 @@ void generateV2Header(const Network::Address::Ip& source_address,
 }
 
 bool generateV2Header(const Network::ProxyProtocolData& proxy_proto_data, Buffer::Instance& out,
-                      bool pass_all_tlvs, const absl::flat_hash_set<uint8_t>& pass_through_tlvs) {
-  uint64_t extension_length = 0;
-  for (auto&& tlv : proxy_proto_data.tlv_vector_) {
```

Comment:
Note: I initially tried expanding the ProxyProtocolData structure to house the custom TLVs, but I ran into several issues trying to modify or inject TLV data after initial construction. I think there's a way to refactor this s.t. we don't have to DI the custom TLVs in this common generateV2Header function, but this implementation was dramatically easier and I didn't think it was worth chasing a refactor at this time.

Reply:
The more I look at this code, I'm wondering whether we could re-use the existing PP listener filter metadata namespace to meet our use case. We just need to populate the PRoxyProtocolData structure later, e.g. in the PP transport socket implementation, in order to do this. I'm not sure what's best here.